test/typings/should_not_compile/index.ts

```diff
@@ -0,0 +1,26 @@
+import {createSelector} from '../../../src/reselect.d.ts';
+import * as common from '../common';
+
+
+const implicitlyTypedSelector = createSelector(
```

Comment:
OK. I've added an issue to get definition tests running on CI, which I'll take. If you get any responses on negative testing strategies, feel free to send another pull request.